### DISCUSSION TITLE
How can I build everything up to project A, but not A itself, like rush.js's --to-except.

### DISCUSSION THREAD

**Discussion by j-xzy** [nrwl/nx#17772] [General]
https://rushjs.io/pages/developer/selecting_subsets/#--to-except

**Comment by AgentEnder** [comment]
There's not a direct way to do this, but you could use a target block similar to this to accomplish it:

> packages/project-a/project.json
```jsonc
{
  // -- more config from project A
  "targets": {
    // -- all the other targets left out for brevity
    "build-dependencies": {
      "executor": "nx:noop",
      "dependsOn": ["^build"]
    }
  }
}
```

This example sets up a target s.t. you'd run something like: `nx run project-a:build-dependencies` to run `build` on all dependencies of `project-a`, but not `build` on `project-a` itself.
